app/lib/frontend/backend.dart

```diff
@@ -126,27 +130,30 @@ class Backend {
     final packageVersionKey = db.emptyKey
         .append(models.Package, id: package)
         .append(models.PackageVersion, id: version);
-    return (await db.lookup([packageVersionKey])).first;
+    return (await db.lookup([packageVersionKey]))
+        .cast<models.PackageVersion>()
```

Comment:
Using `cast()` wraps the list in a new object that does the cast on every access. It doesn't matter so much here, but I believe the recommendation is to cast the object returned by `first` instead:
```dart
return (await db.lookup(...)).first as PackageVersion;
```
https://www.dartlang.org/guides/language/effective-dart/usage#avoid-using-cast